src/ocean/net/server/unix/UnixListener.d

```diff
@@ -226,7 +227,7 @@ public class UnixSocketListener ( CommandHandlerType ) : SelectListener!(
         }
         else
         {
-            log.info("Deleted socket \"{}\"", this.address_pathnul[0 .. $ - 1]);
```

Comment:
I can't see what is wrong with the old code. Can you explain this please? Also I think it would be good to apply the same change to line 224.

Reply:
The `address_pathnul` is a static array of size `sockaddr_un.sun_path.length`. So if it stores a string of three elements, and assuming that `sockaddr_un.sun_path.length` is 12 elements, the old code will print the first three elements, then nul and finally garbage after it (7 characters, because it will not print the last element).

Reply:
> Also I think it would be good to apply the same change to line 224.

:+1:

Reply:
`address_pathnul` is a D char array - i.e a struct with at least length and ptr (https://github.com/sociomantic-tsunami/ocean/blob/bb525d17a230c87e4cf0a7443c3a473266dcb5d9/src/ocean/net/server/unix/UnixListener.d#L134). Its a static array so its length will always be the same (just over 100 bytes on most platforms IIRC).

This array is filled with the path to the socket and a nul byte (https://github.com/sociomantic-tsunami/ocean/blob/bb525d17a230c87e4cf0a7443c3a473266dcb5d9/src/ocean/net/server/unix/UnixListener.d#L165). If we imagine our path was, say 20 bytes long and `address_pathnul` is 100 bytes long then we end up with 20 bytes of path, one byte of nul, and 79 bytes of unitialised memory (though as this is D they will be 0xFF IIRC).

At line 255 we print `address_pathnul[0 .. $ - 1]` (https://github.com/sociomantic-tsunami/ocean/blob/bb525d17a230c87e4cf0a7443c3a473266dcb5d9/src/ocean/net/server/unix/UnixListener.d#L225). so thats bytes 0 to the length of `address_pathnul` - 2. Using our example above then we print bytes 0 to 98 inclusive which will be 20 bytes of our path, one nul byte, and 77 bytes of 0xFFs. What we really want to do is print from byte 0 until the first nul byte. There are many ways to achieve this and this PR does one of them - by creating a new struct where ptr points to `address_pathnul.ptr` but length is set to `strlen(address_pathnul.ptr)`.

You are right that line 224 looks buggy as well - I just didn't hit that one.

Reply:
Sigh, there goes my poor explanation :'(

> though as this is D they will be 0xFF IIRC).

that is correct.

Reply:
Thanks, for that a comprehensive explanation :).